src/_locales/en/messages.json

```diff
@@ -62,14 +62,22 @@
     "message": "Are you sure that you want to reset all options?",
     "description": "The message shown, when the user is prompted to confirm the resetting of all options"
   },
-  "lowContrastRatio": {
-    "message": "The contrast ratio between the QR code color and the background might be too low to be recognized by QR code readers",
+  "lowContrastRatioInfo": {
+    "message": "Your QR code may be hard to scan by some QR code readers because the contrast is too low.",
     "desription": "The message shown when the contrast ratio is too low"
   },
-  "sameColor": {
-    "message": "The background and the foreground color are too similar. You should change this!",
+  "lowContrastRatioWarning": {
+    "message": "The contrast ratio between the QR code color and the background might be too low to be recognized by the most QR code readers",
+    "desription": "The warning shown when the contrast ratio is too low"
+  },
+  "lowContrastRatioError": {
+    "message": "The background and the foreground color are too similar.",
     "description": "The message shown when the color for the QR code and the background are equal"
   },
+  "messageAutoSelectColorButton": {
+    "message": "Calculate best contrast",
```

Comment:
MAybe you can try to shorten this a bit?

Instead of calculate maybe just "Use". (That the JS has to calculate this is a technical detail the user does not even want to know. :smile:)
And instead of contrast maybe contrasting color? Or something else like "best color"? Remember it does not matter to be technically 100% accurate here, the user should just know what happens when they click the button.

Theoretically it could e.g. also be okay for the user to get some color variation (i.e. not 100%ly the contrast color, but a color, which has enough contrast, so the message goes away.)